packages/cubejs-api-gateway/src/gateway.ts

```diff
@@ -210,23 +210,13 @@ class ApiGateway {
 
     app.use(
       `${this.basePath}/graphql`,
-      [
-        ...userMiddlewares,
-        async (req, res, next) => {
-          try {
-            await this.assertApiScope(
-              'graphql',
-              req?.context?.securityContext
-            );
-            if (next) next();
-          } catch (e: unknown) {
-            if (e instanceof CubejsHandlerError) {
-              res.status(e.status).json({ error: e.message });
-            }
```

Comment:
There is no call for `next` when error != `CubejsHandlerError`.
This will lead to unhandled connection.